### PR TITLE
cpu/stm32/pwm: made channel config more flexible

### PR DESCRIPTION
**Pull request by haukepetersen**
alternative to #6350

Now the logical PWM channels used be RIOT are decoupled from the actual used HW CC channels.

### CODE REVIEW

File: boards/stm32f4discovery/include/periph_conf.h [hunk 1]
```diff
@@ -159,21 +159,21 @@ static const pwm_conf_t pwm_config[] = {
...
-    {
-        .dev      = TIM3,
-        .rcc_mask = RCC_APB1ENR_TIM3EN,
-        .pins     = { GPIO_PIN(PORT_B, 4), GPIO_PIN(PORT_B, 5),
-                      GPIO_PIN(PORT_B, 0), GPIO_PIN(PORT_B, 1) },
-        .af       = GPIO_AF2,
-        .chan     = 4,
-        .bus      = APB1
-    }
+    // {
```
Comment (aabadie) [inline boards/stm32f4discovery/include/periph_conf.h]: Why commenting this out ?

**Comment by aabadie** [comment]
Jenkins is failing because of an unrelated reason 
```
/tests-tweetnacl.c:25:23: fatal error: tweetnacl.h: No such file or directory
```

**Comment by kYc0o** [comment]
Let's see what Murdock says...

**Comment by PeterKietzmann** [comment]
Murdock says yes. @aabadie mind to approve?

File: boards/stm32f4discovery/include/periph_conf.h (hunk 1 above)
Comment (haukepetersen) [inline boards/stm32f4discovery/include/periph_conf.h]: This was a debugging leftover, should have been fixed before merging... Will open a separate PR.

Comment (aabadie) [inline boards/stm32f4discovery/include/periph_conf.h]: Fixed by #6404

**Pull request merged**